skbio/tree/_tree.py

```diff
@@ -2705,6 +2705,47 @@ def compare_tip_distances(self, other, sample=None, dist_f=distance_from_r,
         return dist_f(self_matrix, other_matrix)
 
     @experimental(as_of="0.4.0")
+    def bifurcate(self, insert_length=None, remove_singles=True):
+        r""" Restructures tree into a bifurcating tree
+
+        All nodes that have more than 2 children will
+        have additional intermediate nodes injected to ensure that
+        every node has only 2 children
+        Also, any nodes that have a single child will be collapsed
+
+        Parameters
+        ----------
+        remove_singles : bool, optional
+            Removes all nodes that have only a single child and relink
+            of their childrent to their parent node
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> from six import StringIO
+        >>> tree = TreeNode.read(StringIO("((a,b,g,h)c,(d,e)f)root;"))
```

Comment:
minor, and non-blocking, but it would improve the clarity of the example to print `ascii_art` prior to calling `bifurcate` as well as printing `ascii_art` after the call.